api/provider/aws/aws_test/load_balancer_create_test.go

```diff
@@ -0,0 +1,335 @@
+		}
+
+		switch strings.ToLower(port.Protocol) {
+		case "http", "https":
+			listener.SetInstanceProtocol("http")
+		case "tcp", "ssl":
+			listener.SetInstanceProtocol("tcp")
+		}
+
+		listeners[i] = listener
+	}
+
+	createLoadBalancerInput := &elb.CreateLoadBalancerInput{}
+	createLoadBalancerInput.SetLoadBalancerName("l0-test-lb_id")
+	createLoadBalancerInput.SetScheme("internal")
+	createLoadBalancerInput.SetSecurityGroups([]*string{aws.String("env_sg"), aws.String("lb_sg")})
+	createLoadBalancerInput.SetSubnets([]*string{aws.String("priv1"), aws.String("priv2")})
+	createLoadBalancerInput.SetListeners(listeners)
+
+	mockAWS.ELB.EXPECT().
+		CreateLoadBalancer(createLoadBalancerInput).
+		Return(&elb.CreateLoadBalancerOutput{}, nil)
+
+	healthCheck := &elb.HealthCheck{}
+	healthCheck.SetTarget(req.HealthCheck.Target)
+	healthCheck.SetInterval(int64(req.HealthCheck.Interval))
+	healthCheck.SetTimeout(int64(req.HealthCheck.Timeout))
+	healthCheck.SetHealthyThreshold(int64(req.HealthCheck.HealthyThreshold))
+	healthCheck.SetUnhealthyThreshold(int64(req.HealthCheck.UnhealthyThreshold))
+
+	configureHealthCheckInput := &elb.ConfigureHealthCheckInput{}
+	configureHealthCheckInput.SetLoadBalancerName("l0-test-lb_id")
+	configureHealthCheckInput.SetHealthCheck(healthCheck)
+
+	mockAWS.ELB.EXPECT().
+		ConfigureHealthCheck(configureHealthCheckInput).
+		Return(&elb.ConfigureHealthCheckOutput{}, nil)
+
+	target := provider.NewLoadBalancerProvider(mockAWS.Client(), tagStore, mockConfig)
+	result, err := target.Create(req)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, "lb_id", result)
+
+	expectedTags := models.Tags{
+		{
+			EntityID:   "lb_id",
+			EntityType: "load_balancer",
+			Key:        "name",
+			Value:      "lb_name",
+		},
+		{
+			EntityID:   "lb_id",
+			EntityType: "load_balancer",
+			Key:        "environment_id",
+			Value:      "env_id",
+		},
+	}
+
+	for _, tag := range expectedTags {
+		assert.Contains(t, tagStore.Tags(), tag)
+	}
+}
+
+func TestLoadBalancerCreateDefaults(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+	mockConfig.EXPECT().VPC().Return("vpc_id").AnyTimes()
+	mockConfig.EXPECT().Region().Return("region").AnyTimes()
+	mockConfig.EXPECT().AccountID().Return("123456789012")
+	mockConfig.EXPECT().PrivateSubnets().Return([]string{"priv1", "priv2"}).AnyTimes()
+	mockConfig.EXPECT().PublicSubnets().Return([]string{"pub1", "pub2"}).AnyTimes()
+
+	defer provider.SetEntityIDGenerator("lb_id")()
+
+	req := models.CreateLoadBalancerRequest{
+		LoadBalancerName: "lb_name",
+		EnvironmentID:    "env_id",
+		IsPublic:         true,
+		Ports: []models.Port{
+			{
+				ContainerPort: 80,
+				HostPort:      80,
+				Protocol:      "tcp",
+			},
+		},
+		HealthCheck: models.HealthCheck{
```

Comment:
shouldn't `[]models.HealthCheck{}` should be default?

Reply:
For the same reason I have to define at least one `models.Port`, I have to define a `models.HealthCheck` [because the elb.HealthCheck is constructed and needs to access the request's HealthCheck fields](https://github.com/quintilesims/layer0/blob/api-refactor/api/provider/aws/load_balancer_create.go#L103)

Reply:
I think we need to handle a default healthcheck if one isn't specified then. `defaults.go` is a good place for this.